### ISSUE TITLE
How to diagnose a source file without configuration

### ISSUE THREAD

**Issue by andreeis** [microsoft/vscode-cpptools#8379]
Currently when VSCode loads a c/c++ source file that Makefile Tools did not find an IntelliSense configuration for, we log a message in the output channel.
Replace with a diagnostic item in the problems tab.

Inspired from [comment about files without configuration](https://github.com/microsoft/vscode-makefile-tools/issues/71#issuecomment-825762548).

**Comment by Florin-Popescu** [comment]
Could it be configurable by a setting whether or not this particular error would be reported? While it is useful when it indicates that a file you opened is actually never built, I also wouldn't like to pollute the Problems window with errors that show up on every header file, see [#225 in makefile-tools](https://github.com/microsoft/vscode-makefile-tools/issues/225#issue-1046484112).

**Comment by Colengms** [comment]
I don't believe we currently report any warning in this case.  We then use the 'custom browse config' to configure the file (as it includes a compiler, args, standard).

I think we should use a "Configuration Warning".  Currently, we only bring that output channel to front the first time something is logged to it.  We also have an issue tracking adding a setting to optionally never bring it to the front: https://github.com/microsoft/vscode-cpptools/issues/7234
